test/HcalDigiPipelineTest.cxx

```diff
@@ -0,0 +1,259 @@
+
+#include "catch.hpp" //for TEST_CASE, REQUIRE, and other Catch2 macros
+
+#include "DetDescr/HcalID.h" //creating unique cell IDs
+#include "Framework/ConfigurePython.h"
+#include "Framework/EventProcessor.h"
+#include "Framework/Process.h"
+
+namespace hcal {
+namespace test {
+
+/**
+ * Energy deposited by one MIP on average
+ * [MeV]
+ */
+static const double MIP_ENERGY = 4.66;
+
+/**
+ * Conversion between voltage and deposited energy
+ * [MeV/mV]
+ */
+static const double MeV_per_mV = MIP_ENERGY/(5*68); // 0.013 MeV/mV
+
+/**
+ * Maximum percent error that a single hit
+ * can be reconstructed with before failing the test
+ * if above the tot threshold.
+ *
+ * Comparing energy deposited in Silicon that was
+ * "simulated" (input into digitizer) and the reconstructed
+ * energy deposited output by reconstructor.
+ */
+static const double MAX_ENERGY_PERCENT_ERROR_DAQ_TOT_MODE = 2.;
+
+/**
+ * Maximum absolute error that a single hit
+ * can be reconstructed with before failing the test
+ * if below the adc threshold
+ *
+ * Comparing energy deposited in Silicon that was
+ * "simulated" (input into digitizer) and the reconstructed
+ * energy deposited output by reconstructor.
+ */
+static const double MAX_ENERGY_ERROR_DAQ_ADC_MODE = MIP_ENERGY / 2;
+
+/**
+ * Number of sim hits to create.
+ *
+ * In this test, we create one sim hit per event,
+ * run it through the digi pipeline, and then
+ * check it. This parameter tells us how many
+ * sim hits to create and then (combined with
+ * the parameters of HcalFakeSimHits), we know
+ * how "fine-grained" the test is.
+ */
+static const int NUM_TEST_SIM_HITS = 10;
```

Comment:
Why only 10? Feel free to blow this number up! In the ECal, I found some small bugs after have a really fine-grained test.